examples/two_dimensional_turbulence.jl

```diff
@@ -59,7 +59,7 @@ nothing # hide
 # `vorticity_operation` we create a field `ω` to store the result of the operation, and a
 # `Computation` object for coordinate the computation of vorticity and storage in `ω`:
 
-ω = Field(Face, Face, Cell, model.architecture, model.grid)
+ω = Field(Face, Face, Cell, model.architecture, model.grid, TracerBoundaryConditions(model.grid))
```

Comment:
I guess it could go either way in general (eg, `TracerBoundaryConditions` could be correct in other topologies). Still, default might be best here to avoid confusion...

Reply:
Ah so not in this case because we only have default boundary conditions for `CellField` and `*FaceField` but this is a `(Face, Face, Cell)` field so we need to use the `Field` constructor and specify boundary conditions.

I do agree this isn't ideal for an introductory example but not sure what would be more beginner friendly here... (suggestions welcome)

Reply:
I guess in the future we can consider upgrading `Oceananigans.AbstractOperations` so that users don't have to create a new field to store computation results.

This would have the benefit of simplifying user scripts and they won't have to know about the staggered grid.

Reply:
Oh. We should just have defaults for location, topology pairs. Periodic and Flat directions are always periodic. Face, Bounded is NoPenetration. Cell, Bounded is NoFlux. Then defaults are defined for all possible fields. This isn’t what we have?

Reply:
Certainly we can provide that convenience for Computations, that would be easy.

Reply:
Looks like we do have the right defaults; we are just missing a convenience constructor that executes the defaults, here: https://github.com/climate-machine/Oceananigans.jl/blob/b29988d2b5693543a2af2943e85e1a00ab7899c6/src/Fields/field.jl#L77

Reply:
Go ahead and merge — I can fix this later